### ISSUE TITLE
Allow `expect_identical(names(x), names(y))`

### ISSUE THREAD

**Issue by Bisaloo** [r-lib/lintr#1965]
``` r
lintr::lint(
  text = "expect_identical(names(x), names(y))",
  linter = lintr::expect_named_linter()
)
#> <text>:1:1: warning: [expect_named_linter] expect_named(x, n) is better than expect_identical(names(x), n)
#> expect_identical(names(x), names(y))
#> ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<sup>Created on 2023-04-20 with [reprex v2.0.2.9000](https://reprex.tidyverse.org)</sup>

In this specific case, I find `expect_identical(names(x), names(y))` more readable than `expect_named(x, names(y))`. Can we allow this situation in `expect_named_linter()` by checking if the 2nd element of the comparison is another call to `names()`?

In terms of error message, I also believe `expect_identical()` gives a better error message for this specific situation:

``` r
testthat::expect_named(
  cars,
  names(iris)
)
#> Error: Names of `cars` ('speed', 'dist') don't match 'Sepal.Length', 'Sepal.Width', 'Petal.Length', 'Petal.Width', 'Species'

testthat::expect_identical(
  names(cars),
  names(iris)
)
#> Error: names(cars) not identical to names(iris).
#> Lengths differ: 2 is not 5
```

<sup>Created on 2023-04-20 with [reprex v2.0.2.9000](https://reprex.tidyverse.org)</sup>

**Comment by AshesITR** [comment]
Seems like a good idea to me.

Off-Topic: `linter=` in your example code partially matches to `linters=`; you might want to avoid partial matching and use the full argument name instead.

**Comment by MichaelChirico** [comment]
I would prefer something more general, like expect_identical(foo(x), foo(y)) never throws, controlled by an option like allow_matched_calls

**Comment by AshesITR** [comment]
For `expect_named_linter` or in general?
`expect_setequal_linter` comes to mind as not needing this argument, for example.

**Comment by MichaelChirico** [comment]
oh! I think I was confusing myself. thought this logic was going into expect_identical_linter().

anyway, yes, we can imagine other cases like expect_identical(class(x), class(y)).

I'm borderline on whether this should be exposed via an option. my original thinking was requiring expect_named(x, names(y)) is preferable because in most (?) practical cases, it's more like

expect_identical(actual, names(expected))

or

expect_named(MyFunction(x), names(y))

which I still find more appealing

**Comment by Bisaloo** [comment]
> I'm borderline on whether this should be exposed via an option. my original thinking was requiring expect_named(x, names(y)) is preferable because in most (?) practical cases, it's more like
> 
> expect_named(actual, names(expected))
> 
> or
> 
> expect_named(MyFunction(x), names(y))
> 
> which I still find more appealing

I just really like the symmetry of `expect_identical()` in this case :shrug:.

> expect_identical(class(x), class(y))

Yes, fully agreed, it's another case I encountered but didn't open an issue then.
